cycler.py

```diff
@@ -38,6 +38,30 @@
     {'color': 'b', 'linestyle': '-'}
     {'color': 'b', 'linestyle': '--'}
     {'color': 'b', 'linestyle': '-.'}
+
+RememberTheStyle
+================
+
+Helper class for mapping keys -> styles::
+
+   from cycler import RememberTheStyle, cycler
+
+   rts = RememberTheStyle((cycler('c', 'rgb') +
+                           cycler('ls', ['-', '--', ':']))
+
+   rts['cat']
```

Comment:
I really like the getitem interface as it makes use of the natural value by
key lookup that python provides.

I don't think it is any weirder than a defaultdict.  It also allows this
object to be a drop in replacement for a dict anyplace that the user is
currently generating a dict of styles and once used can be converted to a
normal dict with `dd = dict(x)`.

I will add an alias named for getitem.

On Wed, Aug 26, 2015, 11:07 AM Phil Elson <notifications@github.com> wrote:

> In cycler.py
> <https://github.com/matplotlib/cycler/pull/16#discussion_r37993202>:
>
> > @@ -38,6 +38,30 @@
> >      {'color': 'b', 'linestyle': '-'}
> >      {'color': 'b', 'linestyle': '--'}
> >      {'color': 'b', 'linestyle': '-.'}
> > +
> > +RememberTheStyle
> > +================
> > +
> > +Helper class for mapping keys -> styles::
> > +
> > +   from cycler import RememberTheStyle, cycler
> > +
> > +   rts = RememberTheStyle((cycler('c', 'rgb') +
> > +                           cycler('ls', ['-', '--', ':']))
> > +
> > +   rts['cat']
>
> I find the overridden __getitem__ really non-intuitive and would highly
> recommend using a method for this purpose. Something like:
>
> class NamedCycler:
>     def __init__(self, cycler, names=None):
>         self.cycler = cycler
>         if names:
>             self._cache = dict(zip(names, cycler))
>         else:
>             self._cache = {}
>
>     def associated_name(self, key):
>         """
>         Return a cycler item for the given key. If the key hasn't already been seen, the
>         item will come from the next iteration of the cycler.
>
>         """
>         if key not in self._cache:
>             self._cache[key] = next(self.cycler)
>         return self._cache[key]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/cycler/pull/16/files#r37993202>.
>